kind/kind.sh

```diff
@@ -414,59 +471,85 @@ EOF
         fi
 
         CERT_DIR=/usr/local/share/ca-certificates
-        certfile="certs/${registry_server}/client.crt"
+        certfile="$HOME/.registry/certs/${registry_name}/client.crt"
 
         while IFS= read -r container; do
-            log_message "1" "==== Copying ${certfile} to ${container}:${CERT_DIR}"
-            docker cp "$certfile" "${container}:${CERT_DIR}"
+            note_start_task "1" "Copying ${certfile} to ${container}:${CERT_DIR}"
+            ${CRI_COMMAND} cp "$certfile" "${container}:${CERT_DIR}"
+            succeeded "1" "Copying ${certfile} to ${container}:${CERT_DIR}"
 
-            log_message "1" "==== Updating CA certificates in ${container}..."
-            docker exec "$container" update-ca-certificates
+            note_start_task "1" "Updating CA certificates in ${container}..."
+            ${CRI_COMMAND} exec "$container" update-ca-certificates
+            succeeded "1" "Updating CA certificates in ${container}..."
 
-            log_message "1" "==== Restarting containerd"
-            docker exec "$container" systemctl restart containerd
+            note_start_task "1" "Restarting containerd..."
+            ${CRI_COMMAND} exec "$container" systemctl restart containerd
+            succeeded "1" "Restarting containerd..."
         done <<< "$containers"
 
         log_message "1" "Copy the client.crt to the docker cert.d folder"
-        sudo mkdir -p /etc/docker/certs.d/${SERVER_IP}.sslip.io:5000
-        sudo cp $certfile /etc/docker/certs.d/${SERVER_IP}.sslip.io:5000/ca.crt
-        sudo service docker restart
+        if [[ "$OSTYPE" != "darwin"* ]]; then
+            sudo mkdir -p /etc/docker/certs.d/${SERVER_IP}.sslip.io:${REGISTRY_PORT}
+            sudo cp $certfile /etc/docker/certs.d/${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
+        else
```

Comment:
This step can be removed as the else block works on macos or linux ;-)

Reply:
fixed